### PR TITLE
Fix value id and metadata

### PR DESCRIPTION
**Pull request by marcelveldt**
fix value_id creation (propertyKey is optional)
fix typo in atrribute name

### CODE REVIEW

**Comment by marcelveldt** [comment]
@balloob just a small PR with some missing stuff.

File: zwave_js_server/model/value.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ def value_id(node: "Node", event: dict) -> str:
     """Return ID of value."""
     command_class = event.get("commandClass")
     endpoint = event.get("endpoint")
-    property_key_name = event.get("propertyKeyName")
+    property_key_name = event.get("propertyKeyName") or event.get("property")
```
Comment (balloob) [inline zwave_js_server/model/value.py:14]: Can we just always use property? It's important that we stay consistent. 

Do we know why keyName can be None?

**Review by balloob — APPROVED** [review]

File: zwave_js_server/model/value.py (hunk 1 above)
Comment (marcelveldt) [reply zwave_js_server/model/value.py:14]: no we can't as this is for the more complex CC's where one CC can have multiple values, fore example the notification CC

Comment (marcelveldt) [reply zwave_js_server/model/value.py:14]: I'll try to find an example if you want. Ran into it while testing

Comment (marcelveldt) [reply zwave_js_server/model/value.py:14]: Schermafbeelding 2021-01-08 om 10 52 44

Comment (marcelveldt) [reply zwave_js_server/model/value.py:14]: It's a sort of (optional) nested structure

File: zwave_js_server/model/value.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ def value_id(node: "Node", event: dict) -> str:
     """Return ID of value."""
     command_class = event.get("commandClass")
     endpoint = event.get("endpoint")
```
Comment (MartinHjelmare) [inline zwave_js_server/model/value.py:13]: It's weird that we accept None value below in the f-string. Is that really intended?

Comment (marcelveldt) [reply zwave_js_server/model/value.py:13]: property should always be there

Comment (MartinHjelmare) [reply zwave_js_server/model/value.py:13]: If these are not optional we should use `dict[key]`.

Comment (marcelveldt) [reply zwave_js_server/model/value.py:13]: the value id should be:

node_id + endpoint_id + command_class_id + property + optional propertyKey

File: zwave_js_server/model/value.py (hunk 1 above)
Comment (MartinHjelmare) [reply zwave_js_server/model/value.py:14]:
```suggestion
    property_key_name = event.get("propertyKeyName") or event["property"]
```

File: zwave_js_server/model/value.py (hunk 2 above)
Comment (MartinHjelmare) [reply zwave_js_server/model/value.py:13]: Endpoint is optional according to the docs.

Comment (marcelveldt) [reply zwave_js_server/model/value.py:13]: I've also changed commandclass as it is mandatory too

Comment (MartinHjelmare) [reply zwave_js_server/model/value.py:13]: So is it ok to have a string `"None"` in the value id for the endpoint position?

Comment (balloob) [reply zwave_js_server/model/value.py:13]: I think it's fine. We can also do `event.get('endpoint', "")` to just have empty string.

Comment (marcelveldt) [reply zwave_js_server/model/value.py:13]: hmm, cosmetical but maybe we should build the string instead or convert it some other presentation

Comment (balloob) [reply zwave_js_server/model/value.py:13]: It's important the `-` are still there to wrap it.

Comment (marcelveldt) [reply zwave_js_server/model/value.py:13]: Leave it as is ?

Comment (MartinHjelmare) [reply zwave_js_server/model/value.py:13]: At minimum we should document the intention in a docstring or comment.

File: zwave_js_server/model/value.py [hunk 3]
```diff
@@ -9,9 +9,9 @@
 
 def value_id(node: "Node", event: dict) -> str:
     """Return ID of value."""
-    command_class = event.get("commandClass")
-    endpoint = event.get("endpoint")
-    property_key_name = event.get("propertyKeyName")
+    command_class = event["commandClass"]
+    endpoint = event.get("endpoint", "")
```
Comment (MartinHjelmare) [inline zwave_js_server/model/value.py:13]: Should we be explicit instead and have a default like `"no_endpoint"`?

Comment (balloob) [reply zwave_js_server/model/value.py:13]: I think that's fine as the ID is not supposed to be human readable.

Comment (MartinHjelmare) [reply zwave_js_server/model/value.py:13]: Yeah, but we'll still read this when debugging, right? Having a double dash is easy to glance over or misinterpret, I'm thinking.

Comment (marcelveldt) [reply zwave_js_server/model/value.py:13]: I'm fine with some default value like "00" or "no_endpoint".
As far as I can see nodes always have an endpoint.

Comment (marcelveldt) [reply zwave_js_server/model/value.py]: @MartinHjelmare changed default value to 00 and resolved the merge conflict

**Review by MartinHjelmare — APPROVED** [review]
Thanks!

File: zwave_js_server/model/value.py [hunk 4]
```diff
@@ -37,10 +37,9 @@ class ValueDataType(TypedDict, total=False):
 
 def value_id(node: "Node", event_data: ValueDataType) -> str:
     """Return ID of value."""
-    command_class = event_data.get("commandClass")
-    endpoint = event_data.get("endpoint")
-    property_key_name = event_data.get("propertyKeyName")
-    # TODO: Should we really include None items in the f-string below?
+    command_class = event["commandClass"]
+    endpoint = event.get("endpoint", "00")
+    property_key_name = event.get("propertyKeyName") or event["property"]
```
Comment (MartinHjelmare) [inline zwave_js_server/model/value.py:42]:
```suggestion
    command_class = event_data["commandClass"]
    endpoint = event_data.get("endpoint", "00")
    property_key_name = event_data.get("propertyKeyName") or event_data["property"]
```

**Review by MartinHjelmare — APPROVED** [review]

File: zwave_js_server/model/value.py (hunk 4 above)
Comment (marcelveldt) [reply zwave_js_server/model/value.py]: hmm that happened with merging I guess

File: zwave_js_server/model/value.py [hunk 5]
```diff
@@ -87,6 +86,10 @@ def unit(self) -> Optional[str]:
         return self.data.get("unit")
 
     @property
+    def states(self) -> Optional[dict]:
+        """Return (optional) states."""
+        return self.data.get("states")
```
Comment (MartinHjelmare) [inline zwave_js_server/model/value.py:91]: We need to add this item to the TypedDict.

Comment (marcelveldt) [reply zwave_js_server/model/value.py:91]: will you do so or should I ?

Comment (MartinHjelmare) [reply zwave_js_server/model/value.py:92]: I did it now.

**Comment by MartinHjelmare** [comment]
Note that I didn't add all the items in the meta data typed dict in my typing effort. I wasn't completely sure about the items since some items that we expected were also not present in the docs. It would be good to look over all of this at some point.

**Comment by marcelveldt** [comment]
> Note that I didn't add all the items in the meta data typed dict in my typing effort. I wasn't completely sure about the items since some items that we expected were also not present in the docs. It would be good to look over all of this at some point.

That's because values can be CC specific

**Pull request merged**